qa/rpc-tests/pegging.py

```diff
@@ -271,6 +288,46 @@ def write_bitcoin_conf(datadir, rpcport, rpcpass=None, p2p_port=None, connect_po
         if "confirmations" not in tx or tx["confirmations"] == 0:
             raise Exception("Peg-in confirmation has failed.")
 
+    print("Test pegout")
+    test_pegout(bitcoin.getnewaddress(), sidechain)
+
+    print("Test pegout Segwit (P2SH of witness script)")
+    parent_chain_addr = bitcoin.getnewaddress()
+    parent_chain_sw_addr = bitcoin.addwitnessaddress(parent_chain_addr)
+    test_pegout(parent_chain_sw_addr, sidechain)
+
+    print("Test pegout Garbage")
+    parent_chain_addr = "garbage"
+    try:
+        test_pegout(parent_chain_addr, sidechain)
+        raise Exception("A garbage address should fail.")
+    except JSONRPCException as e:
+        assert("Invalid Bitcoin address" in e.error["message"])
+        pass
+
+    print("Test pegout Garbage valid")
+    prev_txid = sidechain.sendtoaddress(sidechain.getnewaddress(), 1)
+    sidechain.generate(1)
+    pegout_hex = 100 * "aaaaaaaaaa"
+    inputs = [{"txid": prev_txid, "vout": 0}]
+    outputs = {"data": pegout_hex}
+    rawtx = sidechain.createrawtransaction(inputs, outputs)
+    raw_pegout = sidechain.decoderawtransaction(rawtx)
+    
+    assert 'vout' in raw_pegout and len(raw_pegout['vout']) > 0
+    pegout_tested = False
+    for output in raw_pegout['vout']:
+        scriptPubKey = output['scriptPubKey']
+        if 'type' in scriptPubKey and scriptPubKey['type'] == 'nulldata':
+            assert ('pegout_hex' in scriptPubKey and 'pegout_asm' in scriptPubKey and 'pegout_type' in scriptPubKey and
+                    'pegout_chain' in scriptPubKey and 'pegout_reqSigs' not in scriptPubKey and 'pegout_addresses' not in scriptPubKey)
+            assert scriptPubKey['pegout_chain'] == 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa01f4'
```

Comment:
why is there `01f4` at the end? Looks like you made 500 bytes worth of `aa` and the first 32 should just be `aa`?

Reply:
yeah, I think it has to do with the conversion to int256_t, but is worth investigating more.

Reply:
The 01f4 prefix seems to change if more data is pushed. It seems to have to do with the pushdata ops needed depending on the size of the data.